sklearn/utils/_array_api.py

```diff
@@ -85,7 +85,7 @@ def yield_namespace_device_dtype_combinations(include_numpy_namespaces=True):
 
         elif array_namespace == "array_api_strict":
             try:
-                import array_api_strict
+                import array_api_strict  # noqa
```

Comment:
```suggestion
                import array_api_strict
```

ruff does not like `# noqa` anymore. Maybe we need `pragma: no cover` instead?

Let's see if it's actually needed first.